File: front/scada-front/src/app/DTOs/AnalogInputDTO.ts
export interface AnalogInputDTO {
    Name: string;
    Description: string;
    Address: string;
    ScanTime: number;
    IsScanning: boolean;
    LowLimit: number;
    HighLimit: number;
    Unit: string
  }

export function createAnalogInputDTO(Name: string, Description: string, Address: string, ScanTime: number, IsScanning: boolean, LowLimit: number, HighLimit: number, Unit: string): AnalogInputDTO {
    return {Name, Description, Address, ScanTime, IsScanning, LowLimit, HighLimit, Unit};
  }

export interface AnalogInputIdDTO {
    Id: number;
    Name: string;
    Description: string;
    Address: string;
    ScanTime: number;
    IsScanning: boolean;
    LowLimit: number;
    HighLimit: number;
    Unit: string
  }

export function createAnalogInputIdDTO(Id: number, Name: string, Description: string, Address: string, ScanTime: number, IsScanning: boolean, LowLimit: number, HighLimit: number, Unit: string): AnalogInputIdDTO {
    return {Id, Name, Description, Address, ScanTime, IsScanning, LowLimit, HighLimit, Unit};
  }

File: back/scada-back/DTOs/AnalogInputDTO.cs
using scada_back.Models;
using System.ComponentModel.DataAnnotations.Schema;

namespace scada_back.DTOs
{
    public class AnalogInputDTO
    {
        public string Name { get; set; }

        public string Description { get; set; }

        public string Function { get; set; }

        public string Address { get; set; }

        public int ScanTime { get; set; }

        public bool IsScanning { get; set; }

        public double LowLimit { get; set; }

        public double HighLimit { get; set; }

        public string Unit { get; set; }
    }
}
